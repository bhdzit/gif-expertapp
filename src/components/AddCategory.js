import React, { useState } from 'react'
import PropTypes from 'prop-types';
function AddCategory({ setcategory }) {

    const [inputValue, setinputValue] = useState('');

    const hanleInputChange = (e) => {

        setinputValue(e.target.value);
    }

    const hanledSumitForm = (e) => {
        e.preventDefault();
        if (inputValue.length > 0) {
            setcategory(cats => [inputValue,...cats]);
            setinputValue("");
        }
    }

    return (
        <form onSubmit={hanledSumitForm}>
            <input type="text" value={inputValue} onChange={hanleInputChange} />
        </form>
    )
}

AddCategory.propTypes={
  setcategory:  PropTypes.func.isRequired
}
export default AddCategory;
