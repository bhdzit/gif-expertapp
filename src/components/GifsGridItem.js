import React from 'react'
import PropTypes from 'prop-types';
export const GifsGridItem = ({url,titel}) => {
   // console.log(url);
    return (
        <div className="card">
            <img src={url} alt={titel}/>   
            <p>{titel}</p>    
        </div>
    )


}

GifsGridItem.propTypes={
    titel:PropTypes.string.isRequired,
    url:PropTypes.string.isRequired,
}


