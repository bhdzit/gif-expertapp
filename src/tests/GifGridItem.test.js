const { shallow } = require("enzyme");
import { GifsGridItem } from '../components/GifsGridItem';


describe('Componente gif', () => {
    const titel="Gif N";
    const url="../ruta/img.jpg";
    test('Render de comoponete', () => {
        let wrapper = shallow(<GifsGridItem url={url} titel={titel} />);
        expect(wrapper).toMatchSnapshot();

    })

})
