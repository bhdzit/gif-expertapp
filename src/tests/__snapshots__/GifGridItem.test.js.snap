// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Componente gif Render de comoponete 1`] = `
<div
  className="card"
>
  <img
    alt="Gif N"
    src="../ruta/img.jpg"
  />
  <p>
    Gif N
  </p>
</div>
`;
